python/simulation.cpp

```diff
@@ -6,24 +8,174 @@
 
 #include "context.hpp"
 #include "error.hpp"
+#include "pyarb.hpp"
 #include "recipe.hpp"
+#include "schedule.hpp"
+
+namespace py = pybind11;
 
 namespace pyarb {
 
-void register_simulation(pybind11::module& m) {
+enum class py_spike_recording {
+    off, local, all
+};
+
+// Wraps an arb::simulation object and in addition manages a set of
+// sampler callbacks for retrieving probe data.
+
+class py_simulation {
```

Comment:
why use the `py_` prefix here, when it is declared in the `pyarb` namespace?
The `_shim` postfix has been used for such wrappers to date.

Reply:
Usage is inconsistent. We have `py_recipe`  and `py_recipe_shim`, for example.

Reply:
Renamed in 14e1d9e